RecoParticleFlow/PFTracking/plugins/HGCalTrackCollectionProducer.cc

```diff
@@ -81,12 +85,17 @@ HGCalTrackCollectionProducer::HGCalTrackCollectionProducer(const edm::ParameterS
                         : reco::TrackBase::highPurity),
       DPtovPtCut_(iConfig.getParameter<std::vector<double> >("DPtOverPtCuts_byTrackAlgo")),
       NHitCut_(iConfig.getParameter<std::vector<unsigned> >("NHitCuts_byTrackAlgo")),
-      useIterTracking_(iConfig.getParameter<bool>("useIterativeTracking")) {
+      useIterTracking_(iConfig.getParameter<bool>("useIterativeTracking")),
+      magneticFieldToken_(esConsumes<edm::Transition::BeginLuminosityBlock>()),
+      tkerGeomToken_(esConsumes<edm::Transition::BeginLuminosityBlock>()) {
   LogDebug("HGCalTrackCollectionProducer")
       << " HGCalTrackCollectionProducer::HGCalTrackCollectionProducer " << std::endl;
 
   const edm::ParameterSet& geoconf = iConfig.getParameterSet("hgcalGeometryNames");
   hgc_names_[0] = geoconf.getParameter<std::string>("HGC_ECAL");
+  for (unsigned i = 0; i < hgcGeometryTokens_.size(); i++) {
```

Comment:
```suggestion
  for (const auto i : hgcGeometryTokens_) {
```

Reply:
In fact, I'm a little confused why you need an `std::array` of length 1 here.

Reply:
This format was kept to allow for future extrapolation to HGCal hadronic sections.